feat: implement secure authentication tests

salopower opened the pull request:
- Added environment variables for test credentials
- Create login steps
- Implemented positive and negative tests

salopower committed:
feat: implement secure authentication tests

- Added environment variables for test credentials
- Create login steps
- Implemented positive and negative tests

vlevkoniuk commented on home_works/home_work_20/package.json:
```diff
@@ -0,0 +1,206 @@
+    "is-extglob": "^2.1.1",
+    "is-glob": "^4.0.3",
+    "is-number": "^7.0.0",
+    "isexe": "^2.0.0",
+    "js-tokens": "^4.0.0",
+    "js-yaml": "^4.1.0",
+    "jsesc": "^3.1.0",
+    "json-buffer": "^3.0.1",
+    "json-parse-even-better-errors": "^2.3.1",
+    "json-schema-traverse": "^0.4.1",
+    "json-stable-stringify-without-jsonify": "^1.0.1",
+    "keyv": "^4.5.4",
+    "levn": "^0.4.1",
+    "lines-and-columns": "^1.2.4",
+    "locate-path": "^6.0.0",
+    "lodash.merge": "^4.6.2",
+    "loupe": "^3.1.3",
+    "magic-string": "^0.30.17",
+    "make-error": "^1.3.6",
```
а нашо стільки пакетів?
Для чого вони?

salopower replied:
Сам не знаю)
Мабуть автоматом якось додалися, бо я їх не встановлював

vlevkoniuk commented on home_works/home_work_20/src/pages/base.page.ts:
```diff
@@ -0,0 +1,46 @@
+        return this.page.locator('//a[@class="oxd-userdropdown-link" and text()=\'Logout\']');
+    }
+
+    // ========== Constructor ==========
```
забув коменти повидаляти? :)

vlevkoniuk commented on home_works/home_work_20/src/steps/orange-hrm-login.step.ts:
```diff
@@ -0,0 +1,52 @@
+});
+
+Then('Dashboard page is displayed', async function (this: OrangeHrmWorld) {
+    const basePage = new BasePage(this.page, this.context);
+    await basePage.isUserLoggedIn();
```
ти маєш наслідувати бейс пейджу і вже з тієї в якій наслідував викликати isUserLoggedIn()
Ось так як зараз - це помилка

vlevkoniuk commented on home_works/home_work_20/src/steps/orange-hrm-login.step.ts:
```diff
@@ -0,0 +1,52 @@
+
+Then('login error message should be shown with text {string}',
+    async function (this: OrangeHrmWorld, errorMessage: string) {
+        const loginPage = new OrangeHRMLoginPage(this.page, this.context);
```
пейджі мають Lazy ініціалізовуватися в World

vlevkoniuk reviewed (changes requested):
1. лишні пакети
2. коменти до коду
3. сторінки мають ініцалізовуватися в World

salopower committed:
fix: fixes according to comments
- Added Lazy initialization of pages in World via getters
- Removed explicit creation of page instances in steps
- Removed unnecessary page imports in the steps file
- Added the ability to access pages through this in steps
- Delete packages that are not in use

vlevkoniuk commented on home_works/home_work_20/src/worlds/orange-hrm.world.ts:
```diff
@@ -17,6 +22,20 @@ export class OrangeHrmWorld extends World{
+            this._basePage = new BasePage(this.page, this.context);
+        }
+        return this._basePage;
+    }
```
в тебе ж базова пейджа живе у OrangeHRMLoginPage , бо ти від неї наслідувався

salopower replied:
Прибрав, дякую)

vlevkoniuk reviewed (commented):
ну так уже краще
в тебе ж базова пейджа живе у OrangeHRMLoginPage , бо ти від неї наслідувався

salopower committed:
fix: remove redundant BasePage init in World since it's inherited by LoginPage
